templates/lb-fargate-service/cf.yml

```diff
@@ -1,168 +1,185 @@
+      Family: !Join ['', [!Ref ProjectName, '-', !Ref EnvName, '-', !Ref AppName]]
+      NetworkMode: awsvpc
+      RequiresCompatibilities:
+        - FARGATE
+      Cpu: !Ref TaskCPU
+      Memory: !Ref TaskMemory
+      ExecutionRoleArn: !Ref {{clean .ProjectName}}{{clean .EnvName}}{{clean .AppName}}ExecutionRole
+      TaskRoleArn: !Ref {{clean .ProjectName}}{{clean .EnvName}}{{clean .AppName}}TaskRole
       ContainerDefinitions:
-      # TODO: support sidecar images
-        - Essential: true
-          # TODO: support images from different repositories
-          Image: !Sub ${Image}
+        - Name: !Ref AppName
+          Image: !Ref ContainerImage
+          PortMappings:
+            - ContainerPort: !Ref ContainerPort {{if .Variables}}
+          Environment:{{range $name, $value := .Variables}}
+          - Name: {{$name}}
+            Value: {{$value}}{{end}}{{end}}{{if .Secrets}}
+          Secrets:{{range $name, $valueFrom := .Secrets}}
+          - Name: {{$name}}
+            ValueFrom: {{$valueFrom}}{{end}}{{end}}
           LogConfiguration:
             LogDriver: awslogs
             Options:
-              awslogs-group: !Ref LogGroup
-              awslogs-stream-prefix: Service
               awslogs-region: !Ref AWS::Region
-          Name: web
-          PortMappings:
-            - ContainerPort: 80
-              Protocol: tcp
-      Cpu: 256
-      ExecutionRoleArn: !GetAtt ExecutionRole.Arn
-      Memory: 512
-      NetworkMode: awsvpc
-      RequiresCompatibilities:
-        - FARGATE
-      TaskRoleArn: !GetAtt TaskRole.Arn
-
-  LogGroup:
-    Type: AWS::Logs::LogGroup
-    UpdateReplacePolicy: Retain
-    DeletionPolicy: Retain
-
-  ExecutionRole:
+              awslogs-group: !Ref {{clean .ProjectName}}{{clean .EnvName}}{{clean .AppName}}LogGroup
+              awslogs-stream-prefix: ecs
+  {{clean .ProjectName}}{{clean .EnvName}}{{clean .AppName}}ExecutionRole:
     Type: AWS::IAM::Role
     Properties:
       AssumeRolePolicyDocument:
         Statement:
-          - Action: sts:AssumeRole
-            Effect: Allow
+          - Effect: Allow
             Principal:
               Service: ecs-tasks.amazonaws.com
-        Version: 2012-10-17
-
-  LogWritingPolicy:
-    Type: AWS::IAM::Policy
+            Action: 'sts:AssumeRole'
+      Policies:
+        - PolicyName: !Join ['', [!Ref ProjectName, '-', !Ref EnvName, '-', !Ref AppName, SecretsPolicy]]
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: 'Allow'
+                Action:
+                  - 'ssm:GetParameters'
+                  - 'secretsmanager:GetSecretValue'
+                  - 'kms:Decrypt'
+                Resource:
+                  - !Sub 'arn:aws:ssm:${AWS::Region}:${AWS::AccountId}:parameter/*'
+                  - !Sub 'arn:aws:secretsmanager:${AWS::Region}:${AWS::AccountId}:secret:*'
+                  - !Sub 'arn:aws:kms:${AWS::Region}:${AWS::AccountId}:key/*'
```

Comment:
I think we can be smarter about this once we have a `secrets` command. Leaving it like this for now.